### PR TITLE
fido authenticators comands

### PR DESCRIPTION
**Pull request by merlokk**
**here:**
* fido commands. now - fido u2f
* some merge with EMV core files
* jansson
 
**TODO**
* resolve conflict: project needs file that needs jansson's private symbols. now i can think only one way: include jansson source to project(
* add exchangeAPDU14a merge to arm and client side

### CODE REVIEW

**Comment by merlokk** [comment]
For mingw it links:
libjansson.a (have jsonp) -> libjansson.dll.a (dont have jsonp) -> msys-jansson-4.dll (dont have jsonp)
looks like it not possible for mingw. maybe it possible for linux, but we need a crossplatform solution
so) maybe someone know how to link .a file in mingw without .dll file?

if not - it needs just to include jansson's sources to the project(

**Comment by merlokk** [comment]
ok, can be merged.
next PR i try to parse der certificate and check signature

**Comment by merlokk** [comment]
now ready)

**Comment by RfidResearchGroup** [comment]
impressive! 
ready to merge?

File: client/fido/additional_ca.c [hunk 1]
```diff
@@ -38,26 +38,26 @@
...
+"MYN/hIQ5ZM5p7EjfcnMG4CtYkNsfNHc0AhBLdq45rnT87q/6O3vUEtNMafbhU6kt\r\n" \
+"hX7Y+9XFN9NpmYxr+ekVY5xOxi8h9JDIgoMP4VB1uS0aunL1IGqrNooL9mmFnL2k\r\n" \
+"LVVee6/VR6C5+KSTCMCWppMuJIZII2v9o4dkoZ8Y7QRjQlLfYzd3qGtKbw7xaF1U\r\n" \
+"sG/5xUb/Btwb2X2g4InpiB/yt/3CpQXpiWX/K4mBvUKiGn05ZsqeY1gx4g0xLBqc\r\n" \
+"U9psmyPzK+Vsgw2jeRQ5JlKDyqE0hebfC1tvFu0CCrJFcw==\r\n" \
+"-----END CERTIFICATE-----\r\n"
 
 /* Concatenation of all additional CA certificates in PEM format if available */
-const char   additional_ca_pem[] = YUBICO_CA; ///GLOBALSIGN_CA
+const char   additional_ca_pem[] = GLOBALSIGN_CA YUBICO_CA;
```
Comment (RfidResearchGroup) [inline client/fido/additional_ca.c]: Will this work?

Comment (merlokk) [reply client/fido/additional_ca.c]: yes. i tested it. and here we can add any CA certificate we need

**Comment by merlokk** [comment]
ready

File: client/fido/additional_ca.c (hunk 1 above)
Comment (merlokk) [reply client/fido/additional_ca.c]: here may be `\r\n` only in begin and end certificates lines.

**Comment by merlokk** [comment]
```
pm3 --> hf fi re -j fi
hf fi re -j fi

data len: 907
User public key: 04 CD xx 70 24 49 0F 1D 57 xx CA 14 17 11 4E D2 80 D1 E6 1E xx 54 36 B4 F1 33 46 1E 88 xx 01 4A 30 53 69 66 B1 6A 2C ...
Key handle[64]: 91 C5 C6 7D A8 50 43 CC BB 1A 23 8E 5F 59 88 6A A9 93 D1 84 72 45 09 31 17 0A E5 43 B3 47 A2 42 FE E2 AA 92 A0 4D 5E 9A 9E D6 ...
DER certificate[704]: 30 82 02 BC 30 82 01 A4 A0 03 02 01 02 02 04 03 AD F0 12 30 ...
Certificate OK.
Hash[72]: 30 46 02 21 00 E4 19 xx F3 84 4F 91 B6 98 FC 26 BA FA A6 BA 59 6D 3A 3B 70 59 85 AC 76 00 15 51 30 27 2F 50 BB 02 21 00 ...
Signature is OK.

auth command:
hf fido auth 91C5C67DA85043CCBB1A238E5F59886AA993D18472450931170AE543B347A242FEE2AA92A04D5E9A9ED6D634A760D90CEB272E78B7395003986D2AA549487FEB
File `E:\ProxSpace\pm3i\client\fi.json` saved.
pm3 --> hf fi au -j fi
hf fi au -j fi
---------------------------------------------------------------
User presence: not verified
Counter: 90
Hash[71]: 30 45 02 21 00 C4 6E D0 A8 0F 9E 9F CE CA 97 06 6C C6 B9 1A 25 DC 1F 81 C6 6F 9C 58 B7 58 8F 48 92 12 A5 43 5E 02 20 20 3E 9A E2 0B C9 78 8B 65 24 46 5C FA 55 16 A5 8A 10 36 42 25 FF E5 AA CF 28 3E D2 BB ED 03 FC
Signature is OK.
File `E:\ProxSpace\pm3i\client\fi.json` saved.
pm3 -->
```

**Comment by RfidResearchGroup** [comment]
Superb work!

**Comment by merlokk** [comment]
Thanks)))))

**Pull request merged**